scipio/src/file_stream.rs

```diff
@@ -775,6 +785,27 @@ impl StreamWriterState {
         self.file_pos + self.buffer_pos as u64
     }
 
+    fn flushed_pos(&self) -> u64 {
+        self.flushed_pos
+    }
+
+    fn adjust_flushed_pos(&mut self, just_written: u64) {
+        self.out_of_order_write_returns.push(just_written);
+        self.out_of_order_write_returns.sort_by_key(|x| Reverse(*x));
```

Comment:
What is the runtime performance impact for this? Can we use binary search instead and insert directly at the right index?

Reply:
That's a good point. I considered a binary search but gave up because sorting is simpler and this array is as big as the number of in-flight buffers. That's usually something like 10. Not worth fretting too much about, in my opinion.

Let me know if you agree / disagree.

Reply:
If this vectors stays small then sorting is fine, no objections.

Reply:
Can you then move from changes requested to approved ?

Reply:
No, that's not part of the requested changes, what I want is for you to split your patch into two. See above :arrow_up: